R CODE FINAL Alexander Shemetev

```diff
@@ -0,0 +1,791 @@
+
+# 3) Create a function 'outlier_capping' that would have vector or column as inputs. The function then 
+#    replaces all right tail outliers (very huge numbers) with 0.99 quantile and all 
+#    left tail outliers (big negative numbers) with 0.01 quantile of the column.
+#    The function should output the same column\vector with processed outliers    
+
+outlier_capping <- function(x){
+  quantiles <- quantile( x, c(.01, .99 ) )
+  x[ x < quantiles[1] ] <- quantiles[1]
+  x[ x > quantiles[2] ] <- quantiles[2]
+  x
+}
+outlier_capping( raw_data$income )
+
+# 4) Create a function 'process_missing' that would have a column\vector as an input. The function then
+#    replaces all missing values with the mean and returns the processed column\vector of data.
+
+
+install.packages("zoo")
+library(zoo)
+na.aggregate(raw_data)
+process_missing <- function(x) replace(x, is.na(x), mean(x, na.rm = TRUE))
+process_missing(raw_data$income)
+
+# 5) Apply the 'var_summary' function to each column of the data. (Use purrr maps or lapply or sapply - whichever you prefer). Any missing data?
+
+sapply(raw_data, var_summary)  
+# Returns descriptive statistics for each column in the dataframe
+
+# 6) If everything is done correctly, there should be 150 NA's in 'default' column. Separate the dataframe into two
+#    bankloans_existing and bankloans_new based on whether default value is present or missing
+
+# Not very clear which dataframe should be named how among these 2 names offered. 
+# let bankloans_existing => NA (for future predictions in model)
+# let bankloans_new => no-NA (that is why they are new - we have all data)
+# I understand the text of this task may be interpreted in reverse as well (no direct meaning in the prescription of the task - I decided to pick up one of them in order not to lose time - the meaning of the analysis will not get changed)
+
+# OUTPUT is 150 NA in the default column 
+#$default[[1]]
+#x 
+#n  missing distinct     Info      Sum     Mean      Gmd 
+#700      150        2    0.579      183   0.2614   0.3867 
+
+
+#bankloans existing  - those observations for which there is a value for default (either 0 or 1)
+#bankloans_new - those observations for which default is NA
+
+
+#bankloans_existing <- subset(raw_data, raw_data$default == is.na(raw_data$default)) 
+#raw_data$def_exists = ifelse(raw_data$default > -1,1,0)
+#sum(raw_data$def_exists)
+bankloans_new <- subset(raw_data, is.na(raw_data$default))
+bankloans_existing <- subset(raw_data, raw_data$default > -1)
+
+# 7) Apply outlier_capping and process_missing on every column of bankloans_existing
+
+sapply(bankloans_existing, process_missing)
+# good
+sapply(bankloans_existing, outlier_capping)
+
+
+# 8) Calculate the correlation matrix with correlations represented by color (google heatmap, there is really simple solution using on
+#    of the packages mentioned in the course)
+matrix_raw <- as.matrix(raw_data)
+heatmap(matrix_raw)
+# Heatmap successfully created
```

Comment:
heatmap is the type of graph, but your input for heatmap is wrong that is why your graph is not readable